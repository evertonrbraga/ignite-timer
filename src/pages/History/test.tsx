import { render, screen } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { defaultTheme } from '../../styles/themes/default'  
import { History } from '.'

describe('<History />', () => {
  it('should snapshot the History component', () => {
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <History />
      </ThemeProvider>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})