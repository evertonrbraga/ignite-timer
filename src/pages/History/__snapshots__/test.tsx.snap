// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<History /> should snapshot the History component 1`] = `
.c0 {
  flex: 1;
  padding: 3.5rem;
  display: flex;
  flex-direction: column;
}

.c0 h1 {
  font-size: 1.5rem;
  color: #E1E1E6;
}

.c1 {
  flex: 1;
  overflow: auto;
  margin-top: 2rem;
}

.c1 table {
  width: 100%;
  border-collapse: collapse;
  min-width: 600px;
}

.c1 table th {
  background: #323238;
  padding: 1rem;
  text-align: left;
  color: #E1E1E6;
  font-size: 0.875rem;
  line-height: 1.6;
}

.c1 table th:first-child {
  border-top-left-radius: 8px;
  padding-left: 1.5rem;
}

.c1 table th:last-child {
  border-top-right-radius: 8px;
  padding-right: 1.5rem;
}

.c1 table td {
  background: #29292E;
  border-top: 4px solid #202024;
  padding: 1rem;
  font-size: 0%.875rem;
  line-height: 1.6;
}

.c1 table td:first-child {
  width: 50%;
  padding-left: 1.5rem;
}

.c1 table td:last-child {
  padding-right: 1.5rem;
}

.c2 {
  display: flex;
  align-items: center;
  gap: 0.5rem;
}

.c2::before {
  content: '';
  width: 0.5rem;
  height: 0.5rem;
  border-radius: 50%;
  background: #AB222E;
}

.c3 {
  display: flex;
  align-items: center;
  gap: 0.5rem;
}

.c3::before {
  content: '';
  width: 0.5rem;
  height: 0.5rem;
  border-radius: 50%;
  background: #FBA94C;
}

.c4 {
  display: flex;
  align-items: center;
  gap: 0.5rem;
}

.c4::before {
  content: '';
  width: 0.5rem;
  height: 0.5rem;
  border-radius: 50%;
  background: #00875F;
}

<main
  class="c0"
>
  <h1>
    Meu histórico
  </h1>
  <div
    class="c1"
  >
    <table>
      <thead>
        <tr>
          <th>
            Tarefa
          </th>
          <th>
            Duração
          </th>
          <th>
            Início
          </th>
          <th>
            Status
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Tarefa
          </td>
          <td>
            20 minutos
          </td>
          <td>
            Há 2 meses
          </td>
          <td>
            <span
              class="c2"
              statuscolor="red"
            >
              Interrompido
            </span>
          </td>
        </tr>
        <tr>
          <td>
            Tarefa
          </td>
          <td>
            20 minutos
          </td>
          <td>
            Há 2 meses
          </td>
          <td>
            <span
              class="c3"
              statuscolor="yellow"
            >
              Em andamento
            </span>
          </td>
        </tr>
        <tr>
          <td>
            Tarefa
          </td>
          <td>
            20 minutos
          </td>
          <td>
            Há 2 meses
          </td>
          <td>
            <span
              class="c4"
              statuscolor="green"
            >
              Concluído
            </span>
          </td>
        </tr>
        <tr>
          <td>
            Tarefa
          </td>
          <td>
            20 minutos
          </td>
          <td>
            Há 2 meses
          </td>
          <td>
            <span
              class="c4"
              statuscolor="green"
            >
              Concluído
            </span>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</main>
`;
